apis/v1alpha1/httproute_types.go

```diff
@@ -377,70 +396,116 @@ type HTTPRequestHeaderFilter struct {
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
 	//
-	// Support: core
+	// Support: Core
 	//
-	// +kubebuilder:validation:MinItems=1
-	To []HTTPForwardToTarget `json:"to"`
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ServiceName *string `json:"serviceName,omitempty"`
 
-	// ExtensionRef is an optional, implementation-specific extension to the
-	// "action" behavior.  The resource may be "configmaps" (use the empty
-	// string for the group) or an implementation-defined resource (for
-	// example, resource "myrouteactions" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "configmaps".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+	// BackendRef is a reference to a backend to forward matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
 	//
-	// Support: custom
+	// Support: Extended
+	//
+	// +optional
+	BackendRef LocalObjectReference `json:"backendRef,omitempty"`
+
+	// Port specifies the destination port number to use for the backend. If
+	// unspecified and a Service object consisting of a single port definition
+	// is the backend, that port will be used. If unspecified and the backend is
+	// a Service object consisting of multiple port definitions, controllers
+	// must set the "InvalidRoutes" status condition on any Gateway that
+	// includes this Route to true.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port *int32 `json:"port,omitempty"`
+
+	// Weight specifies the proportion of traffic forwarded to a backend,
+	// computed as weight/(sum of all weights in this ForwardTo list). Weight is
+	// not a percentage and the sum of weights does not need to equal 100. If
+	// only one backend is specified, 100% of the traffic is forwarded to that
+	// backend. If unspecified, weight defaults to 1.
+	//
+	// Support: Core (HTTPRoute)
+	//
+	// +kubebuilder:default=1
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=10000
+	Weight int32 `json:"weight,omitempty"`
+
+	// Filters defined at this-level should be executed if and only if the
+	// request is being forwarded to the backend defined here.
+	//
+	// Conformance: Filtering support, including core filters, is NOT guaranteed
+	// at this-level. Use Filters in HTTPRouteRule for portable filters across
+	// implementations.
+	//
+	// Support: Customf
 	//
 	// +optional
```

Comment:
Should the list have a min/max length to match other optional lists?

Reply:
Updated, thanks!